projects/hathor-wallet-headless/0001-multisig-create-mint-melt-token.md

```diff
@@ -0,0 +1,324 @@
+- Feature Name: MultiSig to assist the EVM compatible bridge by supporting create, mint and melt token commands
+- Start Date: 2023-06-14
+- RFC PR: (leave this empty)
+- Hathor Issue: (leave this empty)
+- Author: Alex Ruzenhack alex@hathor.network
+
+#### Table of content
+[table-of-content]: #table-of-content
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Current system](#current-system)
+- [Guide-level explanation](#guide-level-explanation)
+- [Drawbacks](#drawbacks)
+- [Rationale and alternatives](#rationale-and-alternatives)
+- [Prior-art](#prior-art)
+- [Unresolved questions](#unresolved-questions)
+- [Future possibilities](#future-possibilities)
+
+# Summary
+[summary]: #summary
+
+It extends the wallet-headless to support create a custom token, mint and melt token, all using a MultiSig wallet. Add the ability to initialize a MultiSig wallet without seed to support the coordinator service in the operation of sign and push transactions. Also, add an endpoint to inspect transaction data with wallet metadata like balance, etc.
+
+# Motivation
+[motivation]: #motivation
+
+It is part of the solution proposed in the [EVM compatible blockchain design](https://github.com/HathorNetwork/rfcs/blob/doc/evm-compatible-brigde/projects/evm-compatible-bridge/design.md#required-features). It aims to enable create, mint and melt tokens using MultiSig wallet and cooperate with the [federation service](https://github.com/HathorNetwork/rfcs/blob/doc/evm-compatible-brigde/projects/evm-compatible-bridge/design.md#required-features) and the coordinator service.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The features we need to add for the MultiSig wallet:
+* Create a custom token
+* Mint a custom token
+* Melt a custom token
+* Inspect transaction with wallet metadata
+* Initialize wallet without seed
+
+## API endpoints design
+
+Taking by example the `POST:/wallet/p2sh/tx-proposal` and `POST:/wallet/p2sh/tx-proposal/get-my-signatures` endpoints we can identify the following semantic:
+
+`/wallet` -- the wallet module
+`/p2sh` -- a submodule representing a domain of wallet
+`/tx-proposal` -- a command to create a new transaction proposal
+`/tx-proposal/get-my-signatures` -- this time `/tx-proposal` assumes the role of a component of p2sh domain in the wallet module, and `/get-my-signatures` is the command to sign the transaction with the wallet private key
+
+In summary, we can have either syntax:
+
+* `/module/domain/command`, or
+* `/module/domain/component/command`
+
+As our intend is to execute commands, the verb `POST` is the most convenient method to call the endpoints.
+
```

Comment:
I'm wondering about the endpoints and if we should reorganize them. Doesn't seem intuitive that we have
`POST /wallet/p2sh/tx-proposal/get-my-signatures`
`POST /wallet/p2sh/tx-proposal/create-token`
`POST /wallet/p2sh/tx-proposal/mint-token`
`POST /wallet/p2sh/tx-proposal/melt-token`

But we use `POST /wallet/p2sh/tx-proposal/` for sending tokens. It seems more intuitive that we have `POST /wallet/p2sh/tx-proposal/send-token`.

We could add this API and continue supporting the current one, but mark as deprecated.

Reply:
Ok! As we already have everything implemented, I can map as a "future possibility" and open an issue to make this change. What do you think?